### PR TITLE
TINY-10056: Make dialog 'x' close button clickable when dialog is busy

### PR DESCRIPTION
**Pull request by shanmen-tiny**
Related Ticket: TINY-10056

Description of Changes:
* Make dialog blocker only blocks the body and footer
* Clicking on the blocker/dialog header wouldn't focus on the first element in dialog (see ticket for more info)
* Added `isBlocked` API to the `Blocking`.;

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

**Review by MitchCTiny — APPROVED** [review]

File: modules/alloy/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## Unreleased
 
+### Added
+- Added new `isBlocked` API to the `Blocking`.
```
Comment (ltrouton) [inline modules/alloy/CHANGELOG.md:10]:
```suggestion
- Added new `isBlocked` API to `Blocking`.
```

File: modules/tinymce/CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 - Fixed an issue that prevented setting the inline dialog `size` to 'medium' or 'normal'. #TINY-10015
+- Fixed an issue that prevented close button to be clicked when dialog is blocked. #TINY-10056
```
Comment (ltrouton) [inline modules/tinymce/CHANGELOG.md:13]:
```suggestion
- Fixed an issue that prevented the close button from being clicked when the dialog was blocked. #TINY-10056
```

File: modules/tinymce/src/themes/silver/test/ts/browser/window/SilverDialogBlockTest.ts [hunk 3]
```diff
@@ -140,6 +143,27 @@ describe('browser.tinymce.themes.silver.window.SilverDialogBlockTest', () => {
         DialogUtils.close(editor);
         I18n.setCode('en');
       });
+
+      it('TINY-10056: Ensure close button is clickable when dialog is blocked', async () => {
+        const editor = hook.editor();
+        const api = DialogUtils.open(editor, dialogSpec, test.params);
+        api.block('Block message');
+        await pClick(editor, closeButtonSelector);
+        store.cAssertEq(`Ensure that it has clicked (${test.label})`, [ 'clicked' ]);
```
Comment (ltrouton) [inline modules/tinymce/src/themes/silver/test/ts/browser/window/SilverDialogBlockTest.ts:152]: I think we should sanity check the dialog actually closed
```suggestion
        store.cAssertEq(`Ensure that it has clicked (${test.label})`, [ 'clicked' ]);
          UiFinder.notExists(SugarBody.body(), 'div[role=dialog]');
```

**Review by ltrouton — APPROVED** [review]

File: modules/alloy/CHANGELOG.md (hunk 1 above)
Comment (TheSpyder) [reply modules/alloy/CHANGELOG.md:10]:
```suggestion
- Added new `isBlocked` API to the `Blocking` behaviour.
```

**Review by TheSpyder — APPROVED** [review]

**Review by ArvinJ-H — APPROVED** [review]

**Pull request merged**